apps/ledger-live-mobile/src/families/celo/Delegations/LabelRight.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import React, { useState, useCallback } from "react";
+import { TouchableOpacity } from "react-native";
+import { useTranslation } from "react-i18next";
+import LText from "../../../components/LText";
+import InfoModal from "../../../modals/Info";
+
+type Props = {
+  disabled?: boolean,
+  onPress: () => void,
+};
+
+export default function DelegationLabelRight({ onPress, disabled }: Props) {
+  const { t } = useTranslation();
+
+  const [disabledModalOpen, setDisabledModalOpen] = useState(false);
+
+  const onClick = useCallback(() => {
+    if (disabled) setDisabledModalOpen(true);
+    else onPress();
+  }, [onPress, disabled]);
+
+  const onCloseModal = useCallback(() => setDisabledModalOpen(false), []);
+
+  return (
+    <TouchableOpacity onPress={onClick}>
+      <LText semiBold color={disabled ? "grey" : "live"}>
+        {t("account.delegation.addDelegation")}
+      </LText>
+      <InfoModal
+        isOpened={!!disabledModalOpen}
+        onClose={onCloseModal}
+        data={[
+          {
+            title: t("cosmos.info.delegationUnavailable.title"),
```

Comment:
Nit: I've been trying to add all these text snippets to the `celo` object. So `celo.info.delegationUnavailable.title`. Or there might be a common text snippet we can use instead here.

Reply:
improved